### ISSUE TITLE
[ENH] improve `sktime/networks` folder structure

### ISSUE THREAD

**Issue by geetu040** [sktime/sktime#6790]
previously discussed here https://github.com/sktime/sktime/issues/6641. This issue is a small part of the large design proposal presented there.

currently all the network files belonging to forecasting or classification, pytorch or tensorflow, are under the same `sktime/networks` folder.
```

sktime/networks
├── __init__.py
├── tests
│   └── test_all_networks.py
├── base.py
├── cinn.py
├── cnn.py
├── cntc.py
├── fcn.py
├── inceptiontime.py
├── lstmfcn_layers.py
├── lstmfcn.py
├── ltsf
│   ├── data
│   ├── __init__.py
│   ├── layers
│   ├── layers.py
│   ├── _ltsf.py
│   ├── models
│   └── utils
├── macnn.py
├── mcdcnn.py
├── mlp.py
├── resnet.py
├── rnn.py
├── tapnet.py
└── transformer.py
```
this should be refactored to something like this
```
networks/
├── forecasting/
│   ├── pytorch/
│   └── tensorflow/
├── classification/
│   ├── pytorch/
│   └── tensorflow/
```
there are no network files for forecasting using tensorflow, but this can be expected in future and classification using pytorch, that I am currently working on.

- this will help better distribute 10+ files in same folder, which is going to increase in future
- it will also enable better easy naming for new files. for example to implement transformer for classification using pytorch, one can easily name it as `networks/classification/pytorch/transformer.py` and there would be no clash with the transformer of forecasting.
- this can be done without any deprecation

**Comment by fnhirwa** [comment]
Hey @geetu040 this was part of the [design document issue #6641](https://github.com/sktime/sktime/issues/6641) I was supposed to do, I just got stuck with it but I think we can take it along with the `deep_learning` estimators and improve their organisation.

**Comment by geetu040** [comment]
> Hey @geetu040 this is related to the [design document issue #6641](https://github.com/sktime/sktime/issues/6641) I was supposed to do, I just got stuck with it but I think we can take it along with the `deep_learning` estimators and improve their organisation.

yes sure, I see thats assigned to you, I was just breaking that into smaller parts since thats a large design change. And we can sure work on this later I myself am working on other items for now.

**Comment by fkiraly** [comment]
small point regarding the proposed design:

* I think we should list multiple options instead of just one
* how would the "by task" design deal with cases where the same network can be used, say, in forecasting and classification? A more common case would be classification and regression

**Comment by fnhirwa** [comment]
> small point regarding the proposed design:
> 
> * I think we should list multiple options instead of just one
> * how would the "by task" design deal with cases where the same network can be used, say, in forecasting and classification? A more common case would be classification and regression
> 
> Alternative design: flat namespace and tags about related learning tasks, type of network, etc.

Yeah I do agree with this as some networks are used to implement forecasting and classification estimators, I suggest that we group the networks into two categories just `pytorch` based and `tensorflow` based

**Comment by geetu040** [comment]
sounds good!
> some networks can be implemented by forecasting and classification estimators.

I know this can happen in the future and already happening here https://github.com/sktime/sktime/pull/6791 but is there an example for this right now in the module? I would like to see.

**Comment by fkiraly** [comment]
the more common example is classification and regression. In fact, almost all `tensorflow` based networks are examples for this dual use.

**Comment by Wvidit** [comment]
Anyone working on this issue?
